src/box/xlog.c

```diff
@@ -732,7 +732,14 @@ xdir_collect_inprogress(struct xdir *xdir)
 			continue;
 
 		char path[PATH_MAX];
+#pragma GCC diagnostic push
+		/*
+		 * We ignore this warning, since file paths are inherently
+		 * limited by `PATH_MAX`.
+		 */
+#pragma GCC diagnostic ignored "-Wformat-truncation"
```

Comment:
I find this warning useless, because it's clearly stated in the `snrpintf` manual page that this function may truncate output. We often truncate output when we write something to the log and this is expected. According to the [gcc manual](https://man7.org/linux/man-pages/man1/gcc.1.html), the warning is "based on heuristics" and "may also result in false positives". Also, the warning isn't implemented in clang. Taking this all into account, I'd rather leave it disabled. I'm leaving this decision up to @alyapunov.

Anyway, using the pragma to suppress the warning doesn't look nice. It'd be better to check the `snprintf` return value instead. I guess we should print an error if the resulting path doesn't fit in the buffer.

Reply:
NB: The warning should disappear if the return value of `snprintf()` is checked. I don't remember precisely where I read it -- likely somewhere in a discussion in the GCC bug tracker.

Reply:
Yeah, we already discussed that actually this warning is a good point, and we should actually check the return value and log an error if the path got truncated: updated the patch accordingly.